bip-xxxx.mediawiki

```diff
@@ -0,0 +1,344 @@
+* Check that the sender's inputs' sequence numbers are unchanged.
+* If the sender's inputs' sequence numbers are all the same, check that the receiver's contributed inputs match those.
+* Check that the sender's own outputs have not been modified, except for paying increased fee, or by simple shuffling.
+* If sender specified <code>additionalfeeoutputindex=</code> (see later), the fee should have been subtracted from the output at the same index in the original PSBT.
+* If <code>minfeerate</code> was specified, check that the estimated fee rate of the payjoin proposal is not less than this value.
+* Check that the sent amount in the payjoin proposal is less than or equal to the sent amount of the original transaction. (Defined as the sum of the inputs' value to be signed minus the sender's ouput change)
+
+If the sent amount in the payjoin proposal is above the amount sent in the original PSBT
+* Check that the additional paid amount has been add paid to the fee.
+* Check that the estimated fee rate of the payjoin proposal is not more than the fee rate of the original PSBT. (fee estimation is hard, so we should allow ~2 satoshi per inputs as margin of error)
+* If <code>maxadditionalfeecontribution=</code> was specified, check the additional paid amount is less than or equal to this amount.
+* If <code>maxadditionalfeecontribution=</code> was not specified, the sender's software should ask an interactive confirmation to the user.
+* <code>maxadditionalfeecontribution=</code> and <code>minfeerate=</code> should be ignored in the [[#spare-change|spare change]] case.
+
+The sender must be careful to only sign the inputs that were present in the original PSBT and nothing else.
+
+Note:
+* The sender should allow the payment output to be modified by the receiver (The receiver may substitute a P2WPKH payment to P2SH payment to increase privacy)
+* The sender must allow the receiver to add outputs.
+* The sender must allow the receiver to not add any input. Useful for the receiver to change the paymout output scriptPubKey type.
+* If no input have been added, the sender's wallet implementation should accept the payjoin proposal, but not mark the transaction as an actual payjoin in the user interface.
+
+Our method of checking the fee allows the receiver and the sender to batch payments in the payjoin transaction.
+It also allows the receiver to pay the fee for batching adding his own outputs.
+
+On top of those check, it is recommended, but not required for the sender to check that:
+* If the sender is making a payjoin with a change (ie, not in the [[#spare-change|spare change]] case), make sure the receiver is paying for any change in the output list.
+
+===Optional parameters===
+
+When the payjoin sender posts the original PSBT to the receiver, he can optionally specify the following HTTP query string parameters:
+
+* <code>v=</code>, the version number of the payjoin protocol that the sender is using. The current version is <code>1</code>.
```

Comment:
Do we think `v=` in the BIP21 uri actually makes sense? This is a version specific to the payjoin feature; what if the URI contains other features which themselves might be versioned? I mean `pjv=` could kinda work but it almost seems like the versioning should be embedded within the `pj=` field. Don't want to open a can of worms there, but `v=` seems .. "off" somehow. No?

Reply:
Sorry email broke thread.

Ok so you're saying the sender would modify the `pj=` string potentially?

Reply:
Either that or just make it `pjver=` not `v=` (really, it's not crucially needed perhaps ... I'm saying it's weird to specify "version" to only a part of what the URI is).
Changing the syntax of what `pj=` contains is a more complex suggestion ... I think it makes more sense (encapsulation), but I'm guessing it's too much of a change so maybe, can of worms, and maybe `pjver=` is fine.
Software that gets written should also make sure to reject stuff that has the version part but not the payjoin part .. which is a trivial point in a way, but illustrates what I'm getting at.

Reply:
I think making the version required fixes any confusion? It's a version agreement protocol, of which "v=1" is the only understood thing currently?

Reply:
So, I'm saying: taken in isolation, what is in this document is already fine. It's unambiguous. What I am worrying about is some kind of interference between this specification (for payjoin, within BIP21) with any other new protocol which also uses BIP21 and might also want to specify a version.
Come to think of it, maybe that question is generalisable: given *any* parameter specification `xyz=`, how do we make sure it doesn't conflict with someone else's use of `xyz=`? Perhaps the argument is: well any server is going to figure out for itself what each possible parameter it serves up, means. So maybe not a problem at all ... just `v=` seems weird in context. Perhaps I'm worrying about nothing, not sure.

Reply:
I suppose if there were ever to be a point of confusion it would be more likely in the client, which might have code supporting other parameter sets than the server does. So better to be as unambiguous as possible, just in case.

Reply:
Ah sorry, yes. That makes sense. `pjver=` as an additional required field or something sounds fine

Reply:
@AdamISZ I think there might be some confusion here.

The `v=` parameter is not added to the BIP21 URI that the receiver initially creates, it's added to the payjoin endpoint request the sender makes to the receiver.